File: python/analysis/derivecolumns.py
import pandas as pd

#############################
## constant / static value ##
#############################

# in-place
person["hungry"] = "yes"

# new copy
hungry = person.assign(hungry="yes")



##################
## derive value ##
##################

def sweet_tooth_and_vowel(s):
    if s["sweet_tooth"] == "yes" and s["name"][0].lower() in ("a", "e", "i", "o", "u"):
        return "yes"
    else:
        return "no"

# in-place
person["hungry"] = person.apply(sweet_tooth_and_vowel, axis=1)

File: r/analysis/derivecolumns.R
library(dplyr)
library(magrittr)

#############################
## constant / static value ##
#############################

# new copy
hungry <- person %>%
  mutate(hungry = "yes")

# in-place
person %<>%
  mutate(hungry = "yes")



##################
## derive value ##
##################

# new copy
hungry <- person %>%
  mutate(hungry = case_when(
    (sweet_tooth == "yes") & (name %>% substr(., 1, 1) %>% tolower() %in% c("a", "e", "i", "o", "u")) ~ "yes",
    TRUE ~ "no"
  ))
